src/ipfsd-daemon.js

```diff
@@ -246,24 +246,44 @@ class Daemon {
       }, this.opts.forceKillTimeout)
     }
 
+    let stoppingErr
+    let subprocessAwaitingErr
+
     try {
       await this.api.stop()
     } catch (err) {
-      if (!killed) {
-        throw err // if was killed then ignore error
-      }
+      // store for possible later use
+      stoppingErr = err
+    }
 
-      daemonLog.info('Daemon was force killed')
+    try {
+      // awaiting subprocesses should allow reaping any handles node vm has open for them
+      await this.subprocess
+    } catch (err) {
+      subprocessAwaitingErr = err
```

Comment:
we already have a catch in the .start for the subprocess

Reply:
Took me a sec to understand, you probably meant this https://github.com/ipfs/js-ipfsd-ctl/blob/2851989eff489ef990e19ef8987d1c3b6c074aba/src/ipfsd-daemon.js#L214

Did some digging/asking around `Promise.prototype.catch` and if one tries this in node repl:

```
(async () => {
    try {
        let p = new Promise(resolve => setTimeout(resolve, 1000)).then(() => { 
            throw new Error("foo")
        });
        p.catch((e) => console.log("caught", e));
        await p
    } catch (e) {
        console.log("leaked", e);
    }
})()
```

The above example hits both console.log's which makes me think that the try-catch is required here.

Is the suggestion to just remove the try/catch from around this or what would it be? I still think the act of `await self.subprocess` here is important.

Reply:
In your example above if you reassign the value of p then the error is only printed once:

```js
(async () => {
    try {
        let p = new Promise(resolve => setTimeout(resolve, 1000)).then(() => { 
            throw new Error("foo")
        });
        p = p.catch((e) => console.log("caught", e));   // <--- reassign
        await p
    } catch (e) {
        console.log("leaked", e);
    }
})()

Promise { <pending> }
> caught Error: foo
    at repl:4:19
    at async repl:7:9
```

..which is why the second try-catch is necessary.

Also this code:

```js
this.subprocess.catch(err => reject(translateError(err))) 
```

will just reject the `ready` promise created & awaited on in `.start` which by this point has already been resolved so the error is effectively swallowed.

Reply:
Yeah, did consider this originally but I understood from MDN that this would make the reassigned `p` (return value from `Promise.prototype.catch`) resolve only if the original promise was rejected, however this didn't seem to be the case when testing this.

So to recap:

 1. Reassign the this.subprocess to be the promise returned by the `this.subprocess.catch(err => reject(...)))` on line 214
 2. Remove second try-catch from `stop`

I'll try to do these later today.